src/server/req_rerun.c

```diff
@@ -141,6 +141,32 @@ force_reque(job *pjob)
 {
 	char  newstate;
 	int  newsubstate;
+	struct batch_request *preq;
+	char hook_msg[HOOK_MSG_SIZE] = {0};
+	int rc;
+
+    /* set job endtime to time_now */
+	pjob->ji_qs.ji_endtime = time_now;
+	set_jattr_l_slim(pjob, JOB_ATR_endtime, pjob->ji_qs.ji_endtime, SET);
+
+	/* Allocate space for the endjob hook event params */
+	preq = alloc_br(PBS_BATCH_EndJob);
+
+	if (preq == NULL) {
+		log_err(PBSE_INTERNAL, __func__, "rq_endjob alloc failed");
+	} else {
+		(preq->rq_ind.rq_end).rq_pjob = pjob;
+
+		/*
+		 * Call process_hooks
+	 	 */
+		rc = process_hooks(preq, hook_msg, sizeof(hook_msg), pbs_python_set_interrupt);
```

Comment:
I think I addressed this to your satisfaction on the forums.  If not, please let me know.

We did realize that no DELETEJOB hook event existed, and that it could be useful, but it was outside the scope of what we were trying to accomplish.